File: LeetCode/2554/solution.py
# 2554. Maximum Number of Integers to Choose From a Range I

class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        banned_set = set(banned)
        sum, cnt = 0, 0

        for i in range(1, n + 1):
            if i in banned_set:
                continue
            if sum + i > maxSum:
                return cnt
            sum += i
            cnt += 1

        return cnt

File: LeetCode/2554/solution.cpp
// 2554. Maximum Number of Integers to Choose From a Range I
class Solution {
public:
    int maxCount(vector<int>& banned, int n, int maxSum) {
        unordered_map<int, bool> mp;
        for(auto a: banned) mp[a] = true;

        int sum = 0, cnt = 0;
        for(int i=1; i<=n; i++){
            if(mp[i]) continue;
            if(sum + i > maxSum) return cnt;
            sum += i;
            cnt++;
        }

        return cnt;
    }
};